src/scales/scale.category.js

```diff
@@ -53,10 +53,25 @@ module.exports = function() {
 
 		getLabelForIndex: function(index, datasetIndex) {
 			var me = this;
-			var data = me.chart.data;
+			var chart = me.chart;
+			var data = chart.data;
 			var isHorizontal = me.isHorizontal();
 
-			if (data.yLabels && !isHorizontal) {
+			// Chart might be undefined or not contain getDatasetMeta function.
+			// This is the case in some tests, and probably only in tests
+			var ds = chart.getDatasetMeta && chart.getDatasetMeta(datasetIndex).controller;
+
+			// Three states here
+			// null:  There is no getValueScaleId in dataset (= not a bar chart)
+			// true:  Scale same as this
+			// false: Value scale is not this scale
+			var isValue = ds && ds.getValueScaleId
```

Comment:
I might write this with two variables, instead of using one variable to represent two things:

```
	var hasValueScale = ds && ds.getValueScaleId;
	var isValueScale = hasValueScale && ds.getValueScaleId() === me.id;

	if (isValueScale || (hasValueScale && !isHorizontal)) {
```

Reply:
Agreed, its more obvious that way

Reply:
Or you could simplify this by:

```javascript
var isValue = ds && ds.getValueScaleId
    ? ds.getValueScaleId() === me.id
    : !isHorizontal;

if (isValue) {
   // ...
}
```

`isValue` is a regular boolean and I find the the check even more obvious.